packages/theme-check-common/src/checks/missing-schema/index.ts

```diff
@@ -17,6 +17,15 @@ export const MissingSchema: LiquidCheckDefinition = {
 
   create(context) {
     let foundSchema = false;
+    const file = context.file;
+    const fileUri = file.uri;
+    const relativePath = context.toRelativePath(fileUri);
+    const isBlockFile = relativePath.startsWith('blocks/');
+
+    if (!isBlockFile) {
+      // Only lint block files as this check is invalid for snippets
+      return {};
+    }
```

Comment:
(nit) The extra variables make this a little more verbose and complicated to reason through. Could be simplified:

```suggestion
    const relativePath = context.toRelativePath(context.file.uri);

    /**
     * Theme app extension blocks are the only types of files that can have a
     * schema defined in them.
     */
    if (!relativePath.startsWith('blocks/')) {
      return {};
    }
```